chucks-pick3-web/angular-app/src/app/components/log-in.component/log-in.component.html

```diff
@@ -1,42 +1,53 @@
 <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
 <div class="main-content container-fluid">
-  <form class="needs-validation" novalidate>
-    <div class="row">
-      <div class=" col-xl-5 mx-auto text-left">
-        <p class="h4 text-center">Login </p>
-        <div class="info-form">
-          <div class="form-row">
-            <label for="defaultRegisterFormUserName">Username</label>
-            <input type="text" minlength="4" name="userName" maxlength="25"
-                   id="defaultRegisterFormUserName"
-                   class="form-control"
-                   placeholder="Username" onchange="checkedUserName(this)" required="required">
-            <div class="invalid-feedback userName"></div>
+<form  class="needs-validation"  #logInForm="ngForm" novalidate>
+  <div class="row">
+    <div class=" col-xl-5 mx-auto text-left">
+      <div *ngIf="errorMessage" class="alert alert-danger">{{errorMessage}}</div>
+      <p class="h4 text-center">Login </p>
+      <div class="info-form">
+        <div class="form-row">
+          <label for="UserName">Username</label>
+          <input id="UserName"
+                 name="userName"
+                 #userName="ngModel"
+                 type="text"
+                 class="form-control" placeholder="User Name"
+                 required ngModel nameValidator>
```

Comment:
nameValidator seems a bit out of place here.  Maybe we should create a usernameValidaor instead.

Reply:
Being completed on separate branch.